### PR TITLE
[CodeGen][X86] LiveRangeShrink: fix increment after end

### PR DESCRIPTION
**Pull request by MuellerMP**
This fixes the infinite loop discovered in #114195. 
Since we skip debug instructions at the start of the loop we do not need to skip them again at the end of the loop.

### CODE REVIEW

**Review by arsenm — COMMENTED** [review]
Needs test

**Comment by nathanchance** [comment]
Thanks, this appears to resolve the issue I reported, but I understand there needs to be a test added before it can be merged.

**Comment by MuellerMP** [comment]
This is how far i could reduce the orginal sample. (Still repros on trunk https://llvm.godbolt.org/z/bh8boEvff)
It requires the debug value + phi setup to trigger this specific issue with debug values at the end of an MBB.

File: llvm/test/CodeGen/X86/lrshrink-debug.ll [hunk 1]
```diff
@@ -0,0 +1,82 @@
+; NOTE: Assertions have been autogenerated by utils/update_llc_test_checks.py UTC_ARGS: --version 4
```
Comment (arsenm) [inline llvm/test/CodeGen/X86/lrshrink-debug.ll:1]: Latest version is 5

**Review by arsenm — APPROVED** [review]

**Pull request merged**